### ISSUE TITLE
Catch unclosed comments

### ISSUE THREAD

**Issue by rayshan** [rvben/rumdl#771]
### Problem or motivation

I use Obsidian. I use `%%` comments heavily to hide content that's useful for AI agents but not for me to read. I noticed that AI agents, e.g. Claude Code + Opus 4.8, starts a comment but sometimes never closes the comment block. An unclosed `%%` silently hides everything to EOF in Obsidian read mode. Obsidian doesn't show an error, and its edit mode looks completely normal.

Currently I verify with `rg -o '%%' <file> | wc -l`. Result must be even. Odd means a comment is bleeding.

I briefly searched rumdl docs and issues, and didn't see a rule to catch unclosed comments.

### Proposed solution

Add a rule to catch unclosed / unterminated comments. Support both inline comments as well as multi-line comment blocks. Support multiple comment formats.

### Feature category

New lint rule

### Example markdown

```markdown
%%This is an Obsidian comment that isn't closed.

<!-- This is an HTML comment that isn't closed.
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Shipped in v0.2.48 as MD086, `unclosed-comments`, and it is on by default.

On your example, run with `--flavor obsidian`:

```
note.md:3:1: [MD086] Unclosed Obsidian comment: '%%' has no matching '%%'
```

and the HTML form reports as `Unclosed HTML comment: '<!--' has no matching '-->'`.

One detail worth knowing: your example reports only the `%%`, not both, because rumdl now models what Obsidian actually does with it. The unclosed `%%` really does swallow the rest of the file, the `<!--` below it included. Close the `%%` and the HTML opener is reported on the next run. That matches what your read mode shows you, and it is a better answer than listing openers the renderer never sees.

It also handles what counting cannot: a `%%` inside a code fence is not a delimiter, so `rg -o '%%' | wc -l` going odd on a documented example is no longer a false alarm.

Thanks @rayshan. A failure the editor renders as normal, with no error anywhere, is the kind worth a default-on rule.

**Comment by rayshan** [comment]
Thank you!
